src/metric/metric.cpp

```diff
@@ -76,6 +76,9 @@ Metric* Metric::CreateMetric(const std::string& type, const Config& config) {
       return new CUDAGammaDevianceMetric(config);
     } else if (type == std::string("tweedie")) {
       return new CUDATweedieMetric(config);
+    } else if (type == std::string("r2")) {
+      Log::Warning("Metric R-squared is not implemented in cuda version. Fall back to evaluation on CPU.");
```

Comment:
```suggestion
      Log::Warning("Metric r2 is not implemented in cuda version. Fall back to evaluation on CPU.");
```

All of the other warning messages in this block match the exact spelling users would pass through `params`. We should follow that with this warning as well.

Reply:
Ah, good catch, thanks for fixing that.